packages/core/src/components/button/buttons.tsx

```diff
@@ -30,20 +32,63 @@ export interface IButtonProps extends IActionProps {
     loading?: boolean;
 }
 
-export class Button extends React.Component<React.HTMLProps<HTMLButtonElement> & IButtonProps, {}> {
+export interface IButtonState {
+    isActive: boolean;
+}
+
+export abstract class BaseButton<T> extends React.Component<React.HTMLProps<T> & IButtonProps, IButtonState> {
```

Comment:
I'd suggest calling this `AbstractButton` and moving it to a separate file. Eventually, we'll want to enable the `max-classes-per-file` TSLint rule (default max # classes is 1) and move `AnchorButton` & `Button` into separate files as well.